.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,41 @@
+name: Run tests
+
+on:
+  push:
+    branches:
+      - main
+      - 'release/*'
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 5
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ubuntu-22.04, ubuntu-24.04, macos-latest, windows-latest]
```

Comment:
Normally, its a good practice to pin the versions of OS and python so that the tests output stay predictable. The probability is low though becaues we are using standar python library here.

Reply:
We want to publish this as a package. The package might be used on a variety of OS and python versions. This matrix aims to cover latest OS versions, in combination with minimal required and latest stable Python versions.

That's my practical approach to make sure the package behaves the same independent of OS or Python version.

Reply:
That also makes sense, I will merge it then.